x-pack/plugins/case/server/routes/api/cases/delete_cases.ts

```diff
@@ -7,10 +7,75 @@
 
 import { schema } from '@kbn/config-schema';
 
+import { SavedObjectsClientContract } from 'src/core/server';
+import { CaseType } from '../../../../common/api';
 import { buildCaseUserActionItem } from '../../../services/user_actions/helpers';
 import { RouteDeps } from '../types';
 import { wrapError } from '../utils';
 import { CASES_URL } from '../../../../common/constants';
+import { CaseServiceSetup } from '../../../services';
+
+async function unremovableCases({
+  caseService,
+  client,
+  ids,
+  force,
+}: {
+  caseService: CaseServiceSetup;
+  client: SavedObjectsClientContract;
+  ids: string[];
+  force: boolean | undefined;
+}): Promise<string[]> {
+  // if the force flag was included then we can skip checking whether the cases are collections and go ahead
+  // and delete them
+  if (force) {
+    return [];
+  }
+
+  const cases = await caseService.getCases({ caseIds: ids, client });
+  const parentCases = cases.saved_objects.filter(
+    /**
+     * getCases will return an array of saved_objects and some can be successful cases where as others
+     * might have failed to find the ID. If it fails to find it, it will set the error field but not
+     * the attributes so check that we didn't receive an error.
+     */
+    (caseObj) => !caseObj.error && caseObj.attributes.type === CaseType.collection
+  );
+
+  return parentCases.map((parentCase) => parentCase.id);
+}
+
+async function deleteSubCases({
+  caseService,
+  client,
+  caseIds,
+}: {
+  caseService: CaseServiceSetup;
+  client: SavedObjectsClientContract;
+  caseIds: string[];
+}) {
+  const subCasesForCaseIds = await caseService.findSubCasesByCaseId({ client, ids: caseIds });
+
+  const subCaseIDs = subCasesForCaseIds.saved_objects.map((subCase) => subCase.id);
+  const commentsForSubCases = await caseService.getAllSubCaseComments({
+    client,
+    id: subCaseIDs,
+  });
+
+  // This shouldn't actually delete anything because all the comments should be deleted when comments are deleted
+  // per case ID
+  await Promise.all(
+    commentsForSubCases.saved_objects.map((commentSO) =>
+      caseService.deleteComment({ client, commentId: commentSO.id })
+    )
+  );
+
+  await Promise.all(
+    subCasesForCaseIds.saved_objects.map((subCaseSO) =>
+      caseService.deleteSubCase(client, subCaseSO.id)
```

Comment:
For another PR: I think maybe is better to first delete the subcase and then the comments if the corresponding subcase has been deleted successfully. This will prevent a use case where all  the comments of a subcase have been deleted and the deletion of the subcase faild.

Reply:
Good point